helm/cosmo/charts/router/templates/deployment.yaml

```diff
@@ -123,6 +128,23 @@ spec:
                   name: {{ include "router.secretName" . }}
                   key: graphApiToken
             {{- end }}
+            - name: PROMETHEUS_ENABLED
+              valueFrom:
+                configMapKeyRef:
+                  name: {{ include "router.fullname" . }}
+                  key: prometheusEnabled
+            {{ if .Values.configuration.prometheus.enabled }}
+            - name: PROMETHEUS_LISTEN_ADDR
+              valueFrom:
+                configMapKeyRef:
+                  name: {{ include "router.fullname" . }}
+                  key: prometheusPort
```

Comment:
Hi, this won't work because the router expects a hostname-port combination. You can construct the `prometheusListenAddr`. similar as with `listenAddress`

Reply:
You're right, when changing to the ConfigMap reference I forgot to build the full address